subsys/bluetooth/shell/bt.c

```diff
@@ -697,6 +697,39 @@ static int cmd_name(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_appearance(const struct shell *sh, size_t argc, char *argv[])
+{
+	uint16_t app;
+	int err;
+	const char *val;
+
+	if (argc == 1) {
+		shell_print(sh, "Bluetooth Appearance: 0x%04x", bt_get_appearance());
+		return 0;
+	}
+
+	if (arc != 2) {
+		shell_error("Not implemented.");
+		return -ENOSYS;
+	}
+
+	val = argv[1];
+	if (strlen(val) != 6 || strncmp(val, "0x", 2) ||
+	    !hex2bin(&val[2], strlen(&val[2]), ((uint8_t *)&app), sizeof(app))) {
+		shell_error(sh, "Argument must be 0x followed by exactly 4 hex digits.");
+		return -EINVAL;
+	};
+	app = sys_be16_to_cpu(app);
```

Comment:
```suggestion
	}
	
	app = sys_be16_to_cpu(app);
```

Reply:
Would it be possible to add this rule to the conformance check?

Reply:
Note that I placed the line like this intentionally. The byte swap is part of the parsing operation. Since we cannot have it in a function, I want to at least group it visually.

Reply:
> Would it be possible to add this rule to the conformance check?

That is what we are hoping for :) This is also part of the on-going code style discussion.